### PR TITLE
2-jws-(MultilayerBUD OrdMlWU OrdMlBU OrdMlBUD OrdMlBUT)

### LINKED ISSUE #2: Multilayer Graphs
Create:
- [ ] _MultilayerWD.gen.m (Blanca)
- [x] #12 (Hang)
- [x] _MultilayerBD.gen.m (Blanca)
- [x] #19

Create:
- [x] _OrdMlWD.gen.m (Hang)
- [x] _OrdMlWU.gen.m (Hang)
- [ ] _OrdMlBD.gen.m (Blanca)
- [ ] _OrdMlBU.gen.m (jws)

Create
- [x] _OrdMxBUD.gen.m (Hang)
- [x] _OrdMxBUT.gen.m (Blanca)
- [x] #11 (jws)
- [x] _MultilayerBUT.gen.m (Blanca)
- [ ] _OrdMlBUD.gen.m (jws)
- [ ] _OrdMlBUT.gen.m (jws)

### PR DESCRIPTION
**Pull request by JiaweiSun0512**
MultilayerWU (copy from hang)
MultilayerBUD
OrdMlWU 
OrdMlBU
OrdMlBUD
OrdMlBUT

### CODE REVIEW

File: braph2genesis/graphs/_MultilayerBUD.gen.m [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+DESCRIPTION (constant, string) is the description of the binary undirected multilayer with fixed densities.
+%%%% ¡default!
+'In a binary undirected multilayer with fixed densities (BUD), the layers are those of binary undirected (BU) multilayer graphs derived from the same weighted supra-connectivity matrices binarized at different densities. The supra-connectivity matrix has a number of partitions equal to the number of densities.'
+
+%%% ¡prop!
+TEMPLATE (parameter, item) is the template of the binary undirected multilayer with fixed densities.
+%%% ¡_prop!
+% % % TEMPLATE (parameter, item) is the graph template to set the graph and measure parameters.
+% % % %%%% ¡_settings!
+% % % 'MultilayerBUD'
```
Comment (iok-ui) [inline braph2genesis/graphs/_MultilayerBUD.gen.m:30]: do these comment need to be erased?

File: braph2genesis/graphs/_MultilayerBUD.gen.m [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        density = densities(i);
+        layer = 1;
+        for j = (i - 1) * L + 1:1:i * L
+            A{j, j} = dediagonalize(binarize(A_WU{layer, layer}, 'density', density));
+            layer = layer + 1;
+        end
+    end
+end
+
+value = A;
```
Comment (iok-ui) [inline braph2genesis/graphs/_MultilayerBUD.gen.m:114]: leave a blank line between line 111 and  line 112

File: braph2genesis/graphs/_MultilayerWU.gen.m [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+ATTEMPTSPEREDGE (parameter, scalar) is the attempts to rewire each edge.
+%%%% ¡_default!
+5
+
+%%% ¡_prop!
+NUMBEROFWEIGHTS (parameter, scalar) specifies the number of weights sorted at the same time.
+%%%% ¡_default!
+10
+
+%% ¡_methods!
```
Comment (iok-ui) [inline braph2genesis/graphs/_MultilayerWU.gen.m]: why these props are after test? Usually the test is the last session of an element script

File: braph2genesis/graphs/_MultilayerWU.gen.m [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+% B3 = 10 * rand(randi(10) + 5) - 5; % random number in U(-5, 5)
+% B = {
+%     B1                           rand(size(B1,1),size(B2,2))  rand(size(B1,1),size(B3,2))
+%     rand(size(B1,1),size(B2,2))' B2                           rand(size(B2,1),size(B3,2))
+%     rand(size(B1,1),size(B3,2))' rand(size(B2,1),size(B3,2))' B3
+%     };
+% 
+% g0 = MultilayerWU('B', B); % 'SEMIPOSITIVIZE_RULE', 'zero', 'STANDARDIZE_RULE', 'threshold'
+% 
+% A1 = B1;
```
Comment (iok-ui) [inline braph2genesis/graphs/_MultilayerWU.gen.m]: If the test for Symmetrize, Semipositivize and Standardize Rules, then these comments should be erased

**Review by iok-ui — CHANGES_REQUESTED** [review]

**Comment by egolol** [comment]
there's a lot of white spaces, other than it's fine.

**Review by egolol — APPROVED** [review]

File: braph2genesis/graphs/_OrdMlBUT.gen.m [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+%%% ¡description!
+In a multilayer binary undirected with fixed thresholds (BUT) graph, layers 
+ consist of binary undirected (BU) multilayer graphs derived from the same 
+ weighted supra-connectivity matrices binarized at different thresholds. Layers 
+ within the binary undirected (BU) multilayer graphs could have different number of nodes 
+ with within-layer binary undirected edges. Edges can be either 0 (absence of connection) 
+ or 1 (existence of connection).
+The supra-connectivity matrix has a number of partitions equal to the number of thresholds.
+The layers are connected in an ordinal fashion, i.e., only consecutive layers are connected.
```
Comment (annacanal) [inline braph2genesis/graphs/_OrdMlBUT.gen.m]: this description is wrong, please correct

File: braph2genesis/graphs/_OrdMlWU.gen.m [hunk 6]
```diff
@@ -0,0 +1,259 @@
+%% ¡header!
+OrdMlWU < Graph (g, ordinal multilayer weighted undirected graph) is an ordinal multilayer weighted undirected graph.
+
+%%% ¡description!
+In an ordinal multilayer weighted undirected (WU) graph, layers have different number 
+ of nodes with within-layer weighted undirected edges, associated with a real 
+ number between 0 and 1 and indicating the strength of the connection.
+The connectivity matrices are symmetric.
+There are connections between layers connecting the corresponding nodes.
+The layers are connected in an ordinal fashion, i.e., only consecutive layers are connected.
```
Comment (annacanal) [inline braph2genesis/graphs/_OrdMlWU.gen.m]: this description is wrong, please correct based on the updated in this PR: https://github.com/giovannivolpe/BRAPH-2-Matlab-beta/pull/9

File: braph2genesis/graphs/_MultilayerWU.gen.m [hunk 7]
```diff
@@ -0,0 +1,514 @@
+%% ¡header!
+MultilayerWU < Graph (g, multilayer weighted undirected graph) is a multilayer weighted undirected graph.
+
+%%% ¡description!
+In a multilayer weighted undirected (WU) graph, layers have different number 
+ of nodes with within-layer weighted undirected edges, associated with a real 
+ number between 0 and 1 and indicating the strength of the connection.
+The connectivity matrices are symmetric.
+There are connections between layers connecting the corresponding nodes.
```
Comment (annacanal) [inline braph2genesis/graphs/_MultilayerWU.gen.m]: you need to update your branch based on this one: https://github.com/giovannivolpe/BRAPH-2-Matlab-beta/pull/9
Descriptions are wrong, you have not updated them

**Review by annacanal — CHANGES_REQUESTED** [review]
You need to update your branch and get the new descriptions, and other changes:
https://github.com/giovannivolpe/BRAPH-2-Matlab-beta/pull/9

**Review by annacanal — APPROVED** [review]

**Review by bzufiriaki — APPROVED** [review]

**Pull request merged**